### PR TITLE
Add REPLACE INTO support

### PR DESCRIPTION
**Pull request by gephaest**

### CODE REVIEW

**Comment by lichtner** [comment]
thanks, I have some notes:
- it is almost same as UpdateQuery, may be both can extend same parent class something like (e.g. ExecQuery)
- replace have also VALUES clause it can be alias for set 
- each feature is covered by tests for common use cases

**Comment by gephaest** [comment]
Ok, fix it in a few days

**Comment by gephaest** [comment]
Ooops, PHP 5.3 doesn't support traits. 
In my opinion, it's good idea to support only PHP 5.4+ versions because 5.3 is no longer supported since 14 Aug 2014 ([proof] (http://php.net/eol.php)).

PHP 5.4 have killer features:
- traits
- short array syntax

**Comment by lichtner** [comment]
I know traits are cool but it is possible to do it without them. Now fluentpdo support also archaic 5.1+. To be honest we still use php 5.3 on production server for some other reason.
I am not sure if it is good idea drop support for php 5.3....

**Comment by gephaest** [comment]
Oh, It's ok.
I replaced trait to helper class, but I'm not sure that elegant way.

File: FluentPDO/SetStatement.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+class SetStatement {
+    private $setStatement, $parameters;
+
+    public function set($fieldOrArray, $value = false) {
+        if (!$fieldOrArray) {
+            return [];
```
Comment (alexantr) [inline FluentPDO/SetStatement.php]: short array syntax not supported in 5.3

File: FluentPDO/ReplaceQuery.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
+<?php
+/** REPLACE INTO query builder
+ */
+class ReplaceQuery extends BaseQuery {
+    private $setStatement;
+
+    public function __construct(FluentPDO $fpdo, $table) {
```
Comment (hoshsadiq) [inline FluentPDO/ReplaceQuery.php]: missing phpdoc

File: FluentPDO/ReplaceQuery.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
+<?php
+/** REPLACE INTO query builder
+ */
+class ReplaceQuery extends BaseQuery {
+    private $setStatement;
```
Comment (hoshsadiq) [inline FluentPDO/ReplaceQuery.php]: missing phpdoc

File: FluentPDO/SetStatement.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?php
+class SetStatement {
+    private $setStatement, $parameters;
```
Comment (hoshsadiq) [inline FluentPDO/SetStatement.php]: missing phpdoc

File: FluentPDO/SetStatement.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    public function set($fieldOrArray, $value = false) {
```
Comment (hoshsadiq) [inline FluentPDO/SetStatement.php]: missing phpdoc

File: FluentPDO/SetStatement.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                foreach ($fieldOrArray as $field => $value) {
+                    $this->setStatement[$field] = $value;
+                }
+            }
+        }
+        return $this->setStatement;
+    }
+
+
+    public function getClauseSet() {
```
Comment (hoshsadiq) [inline FluentPDO/SetStatement.php]: missing phpdoc

**Comment by cbornhoft** [comment]
This would be better left to writing by hand than to use a query builder. Our goal is to keep FluentPDO as lean and fast as possible so as not to get in the way of query speeds too much. Adding an extra two files for a very rarely used method does not meet this requirement.

**Pull request closed without merging**